src/components/DateInput/DateInput.tsx

```diff
@@ -216,11 +230,16 @@ const DateInput: React.FC<DateInputProps & Omit<TextInputProps, 'value' | 'onCha
             <Button onClick={onCancel} size="medium" variantColor="gray">
               Cancel
             </Button>
-            <Button disabled={!currentDate} onClick={onApply} size="medium">
+            <Button disabled={!currentDate && disableReset} onClick={onApply} size="medium">
```

Comment:
same thoughts as above regarding the use of `disableReset` for this kind of check